src/wallet/scriptpubkeyman.cpp

```diff
@@ -1769,9 +1763,13 @@ bool DescriptorScriptPubKeyMan::Encrypt(const CKeyingMaterial& master_key, Walle
 bool DescriptorScriptPubKeyMan::GetReservedDestination(const OutputType type, bool internal, CTxDestination& address, int64_t& index, CKeyPool& keypool, bilingual_str& error)
```

Comment:
In 9a0a4024f0fdd2ec598613fa8ea55683115a88f5 "wallet: refactor GetNewDestination, use BResult"

It would be nice if this function got the same treatment too. But that can also be done in a followup to avoid expanding the scope of this PR too far.

Reply:
yeah, better to leave it for a follow-up PR. Once we get this one merged, other PRs will be able to use the new class in different workflows independently.